galvan/src/main/java/org/terracotta/testing/master/GalvanStateInterlock.java

```diff
@@ -229,14 +241,13 @@ public ServerProcess getOneTerminatedServer() throws GalvanFailureException {
   public void serverBecameActive(ServerProcess server) {
     synchronized (this.sharedLockState) {
       localAssert(null == this.activeServer, server);
-      boolean didRemove = this.unknownRunningServers.remove(server);
-      boolean wasPassive = false;
-      if (!didRemove) {
-        didRemove = this.passiveServers.remove(server);
-        wasPassive = true;
-      }
-      this.logger.output("serverBecameActive: " + server + " was passive:" + wasPassive);
-      localAssert(didRemove, server);
+      boolean wasUnknown = this.unknownRunningServers.remove(server);
+      boolean wasPassive = this.passiveServers.remove(server);
+      boolean wasDiagnostic = this.diagnosticServers.remove(server);
+      String previousState = wasUnknown ? "unknown" : wasPassive ? "passive" : "diagnostic";
```

Comment:
shouldn't unknown be the default instead ? to avoid reporting by mistake something in diagnostic that is not ?

Reply:
or, even better, to catch errors, do a sort of switch case or throw an assertion error if it is something else unsopported, so that we we know if the case happens on day ?

Reply:
This code is removed now.